### PR TITLE
owner: unify the naming of named return value in new changefeed.

### LINKED ISSUE #2389: Changefeed with sink-uri=kafka status become failed after all PD restart
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

- 2x capture, 2x changefeed:
Starting component `cdc`: /root/.tiup/components/cdc/v4.0.14/cdc cli changefeed list --pd=http://172.16.6.28:2379
[
  {
    "id": "kafka-task-11",
    "summary": {
      "state": "normal",
      "tso": 426608082704400445,
      "checkpoint": "2021-07-27 18:11:26.586",
      "error": null
    }
  },
  {
    "id": "replication-task-11",
    "summary": {
      "state": "normal",
      "tso": 426608707681910785,
      "checkpoint": "2021-07-27 18:51:10.686",
      "error": null
    }
  }

- Restart PD: $tiup cluster restart 360UP -R pd

2. What did you expect to see?

- No any error

3. What did you see instead?

- Changefeed with sink-uri=kafka status become failed after all PD restart
   Starting component `cdc`: /root/.tiup/components/cdc/v4.0.14/cdc cli changefeed list --pd=http://172.16.6.28:2379
[
  {
    "id": "replication-task-11",
    "summary": {
      "state": "normal",
      "tso": 426608859290271791,
      "checkpoint": "2021-07-27 19:00:49.026",
      "error": null
    }
  },
  {
    "id": "kafka-task-11",
    "summary": {
      "state": "failed",
      "tso": 426608874219372599,
      "checkpoint": "2021-07-27 19:01:45.976",
      "error": {
        "addr": "172.16.6.32:8300",
        "code": "CDC-owner-1001",
        "message": "rpc error: code = Unknown desc = rpc error: code = Unavailable desc = not leader"
      }
    }
  }
]

- Check cf status again, the failed kafka-task-11 checkpoint is still updating...
Starting component `cdc`: /root/.tiup/components/cdc/v4.0.14/cdc cli changefeed list --pd=http://172.16.6.28:2379
[
  {
    "id": "kafka-task-11",
    "summary": {
      "state": "failed",
      "tso": 426609068572934145,
      "checkpoint": "2021-07-27 19:14:07.376",
      "error": {
        "addr": "172.16.6.32:8300",
        "code": "CDC-owner-1001",
        "message": "rpc error: code = Unknown desc = rpc error: code = Unavailable desc = not leader"
      }
    }
  },
  {
    "id": "replication-task-11",
    "summary": {
      "state": "normal",
      "tso": 426609068677791745,
      "checkpoint": "2021-07-27 19:14:07.776",
      "error": null
    }
  }

4. Versions of the cluster

    - Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

        ```
       4.0.14
        ```

    - TiCDC version (execute `cdc version`):

        ```
        ["Welcome to Change Data Capture (CDC)"] [release-version=v4.0.14] [git-hash=5a7851967f686da896b45acd3f3e968bfe53d6bd] [git-branch=heads/refs/tags/v4.0.14]
        ```

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve? 

In the beginning, this PR try to fix #2389, but it may already be fixed by #2370, this needs to be tested on release-4.0 and other branches.

But When I walking the code, I think #942 does not work properly, because `resultErr` was not written, which was always nil.

### What is changed and how it works?

* use err as a function level variable.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

Side effects

Related changes

### Release note 

```release-note
fix memory leak which may happen in create new changefeed.
```

### CODE REVIEW

**Comment by amyangfei** [comment]
good catch!

File: cdc/owner.go [hunk 1]
```diff
@@ -296,11 +296,11 @@ func (o *Owner) newChangeFeed(
 	processorsInfos model.ProcessorsInfos,
 	taskPositions map[string]*model.TaskPosition,
 	info *model.ChangeFeedInfo,
-	checkpointTs uint64) (cf *changeFeed, resultErr error) {
```
Comment (amyangfei) [inline cdc/owner.go:299]: But if the function exits with `return nil, err`, the `resultErr` will be assigned to value `err`

Comment (3AceShowHand) [reply cdc/owner.go:299]: got it, it would be better to make it explicit ?

Comment (amyangfei) [reply cdc/owner.go:299]: Yes, explicit is better
```

**Review by ben1009 — APPROVED** [review]

**Pull request closed without merging**